apps/ledger-live-desktop/src/newArch/features/ModularDrawer/components/Search/useSearch.ts

```diff
@@ -0,0 +1,67 @@
+import React, { useCallback, useMemo, useState } from "react";
+import Fuse from "fuse.js";
+import { getEnv } from "@ledgerhq/live-env";
+import { track } from "~/renderer/analytics/segment";
+import { CryptoOrTokenCurrency } from "@ledgerhq/types-cryptoassets";
+
+export type SearchProps = {
+  setItemsToDisplay: React.Dispatch<React.SetStateAction<CryptoOrTokenCurrency[]>>;
+  source: string;
+  flow: string;
+  items: CryptoOrTokenCurrency[];
+};
+
+export type SearchResult = {
+  handleSearch: (query: string) => void;
+  trackSearch: (current: string, previous: string) => void;
+  searchQuery?: string;
+};
+
+const FUZE_OPTIONS = {
+  includeScore: false,
+  threshold: 0.1,
+  keys: getEnv("CRYPTO_ASSET_SEARCH_KEYS"),
+  shouldSort: false,
+};
+
+export const useSearch = ({
+  setItemsToDisplay,
+  items,
+  source,
+  flow,
+}: SearchProps): SearchResult => {
+  const [searchQuery, setSearchQuery] = useState("");
+
+  const fuse = useMemo(() => new Fuse(items, FUZE_OPTIONS), [items]);
+
+  const trackSearch = useCallback(
+    (current: string, previous: string) => {
+      const query = current.trim();
+      const prevQuery = previous.trim();
+      if (query === prevQuery) return;
+
+      if (query && query.trim() !== "") {
```

Comment:
For clarity lets just have one early return:
```suggestion
      if (!query || query === prevQuery) return;

```
Also query doesn't need to be trimmed twice

Reply:
done ✅